File: it/test-basic/expected/suite/java/test/Entry.java
package test;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.Objects;
import java.util.Optional;

public class Entry {
  private final Optional<Foo> foo;

  @JsonCreator
  public Entry(
    @JsonProperty("foo") final Optional<Foo> foo
  ) {
    Objects.requireNonNull(foo, "foo");
    this.foo = foo;
  }

  public Optional<Foo> getFoo() {
    return this.foo;
  }

  @Override
  public int hashCode() {
    int result = 1;
    result = result * 31 + this.foo.hashCode();
    return result;
  }

  @Override
  public boolean equals(final Object other) {
    if (other == null) {
      return false;
    }

    if (!(other instanceof Entry)) {
      return false;
    }

    @SuppressWarnings("unchecked")
    final Entry o = (Entry) other;

    if (!this.foo.equals(o.foo)) {
      return false;
    }

    return true;
  }

  @Override
  public String toString() {
    final StringBuilder b = new StringBuilder();

    b.append("Entry");
    b.append("(");
    b.append("foo=");
    b.append(this.foo.toString());
    b.append(")");

    return b.toString();
  }

  public static class Builder {
    private Optional<Foo> foo = Optional.empty();

    public Builder foo(final Foo foo) {
      this.foo = Optional.of(foo);
      return this;
    }

    public Entry build() {
      final Optional<Foo> foo = this.foo;

      return new Entry(foo);
    }
  }
}

File: it/test-basic/expected/suite/csharp/Test/Entry.cs
using Newtonsoft.Json;
using System;
using System.Text;

namespace Test {
  [JsonObject(ItemNullValueHandling = NullValueHandling.Ignore)]
  public class Entry {
    /// <summary>
    /// The foo field.
    /// </summary>
    [JsonProperty("foo")]
    public Foo foo {
      get;
    }

    [JsonConstructor]
    public Entry(
      [JsonProperty("foo")] Foo foo
    ) {
      this.foo = foo;
    }

    public override Int32 GetHashCode() {
      Int32 result = 1;
      result = result * 31 + this.foo.GetHashCode();
      return result;
    }

    public override Boolean Equals(Object other) {
      Entry o = other as Entry;

      if (o == null) {
        return false;
      }

      if (this.foo == null) {
        if (o.foo != null) {
          return false;
        }
      } else {
        if (!this.foo.Equals(o.foo)) {
          return false;
        }
      }

      return true;
    }

    public override String ToString() {
      StringBuilder b = new StringBuilder();

      b.Append("Entry");
      b.Append("(");
      b.Append("foo=");
      b.Append(this.foo);
      b.Append(")");

      return b.ToString();
    }
  }
}
